### ISSUE TITLE
C++ Intellisence engine does not work. 

### ISSUE THREAD

**Issue by ArtBlnd** [microsoft/vscode-cpptools#893]
Engine just keep emits "Initializing..." Like [this](https://puu.sh/wNcF4/20779c3da7.png)
also, seems [C/C++ Extensions For Visual Studio Code](https://puu.sh/wNcKg/dca18467ef.png) processor doesn't even load symbols.

I already tried clean uninstalling VSCode, Exts includes symbol databases.

**Comment by sean-mcmanus** [comment]
0.12.1, right? Did it work with 0.12.0? What OS are you on? Do you see any messages in the Problems pane? The "Initializing..." message only occurs if the Tag Parser (symbol) database has not been initialized yet (which is expected to be a really quick operation that isn't expected to fail, unless it can't create the database file). You might try changing your browse.databaseFilename to something like "${workspaceRoot}/.vscode/browse.vc.db" to see if that file is getting generated at all.

**Comment by ArtBlnd** [comment]
I tried it already( that changing db name) and it is already generated. 
also, I don't see any problems in pane too.

**Comment by sean-mcmanus** [comment]
We need more repro info. What is the CPU usage of the Microsoft.VSCode.CPP.Extension process? A dmp of the process might help. Are you able to get it to work with a small folder with one C++ file?

**Comment by ArtBlnd** [comment]
absolutely zero usage.

**Comment by ArtBlnd** [comment]
can you tell me how do I dump "VSCode.CPP.Extension" process?
I don't know how do I dump, it doesn't even emits any crashes, or errors.
also no errors on VSCode debug console, so how should I dump?

**Comment by ArtBlnd** [comment]
```
- VSCode Version: Code 1.14.1 (2648980a697a4c8fb5777dcfb2ab110cec8a2f58, 2017-07-13T19:18:47.188Z)
- OS Version: Windows_NT ia32 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
language-x86-64-assembly|13x|2.2.11
Doxygen|bbe|1.0.0
Gradle|caz|1.1.2
javadebugger|don|0.1.2
python|don|0.6.7
tslint|eg2|0.16.0
tortoise-svn|fan|0.1.1
solidity|Jua|0.0.22
share|lyu|0.3.1
cpptools|ms-|0.12.1
csharp|ms-|1.11.0
java|red|0.7.0
git-add-remote|sam|0.0.2
cmake|twx|0.0.17
vscode-lldb|vad|0.5.2
cmake-tools|vec|0.10.2

(2 theme extensions excluded)
```

Here is my environment information generated from VSCode.

**Comment by sean-mcmanus** [comment]
You can get a dmp via launching `C:\Windows\SysWow64\Taskmgr.exe` (making sure the 64-bit one in System32 is not running) and right-clicking on the Microsoft.VSCode.CPP.Extension.exe process and selecting "Create dmp file". You can send the dmp to c_cpp_support@microsoft.com . You could also set the environment variables to something like `VSCODE_CPP_LOGDIR=C:\Logs` and ~~`VSCODE_CPP_LOGFILE_LEVEL=9`~~ `VSCODE_CPP_LOGFILE_LEVEL=7` to turn on logging. You might also try disabling your other extensions to make sure they're not interfering some how.

**Comment by bobbrow** [comment]
log level 9 is too much.  7 is all you need.

**Comment by ArtBlnd** [comment]
Okay.

**Comment by ArtBlnd** [comment]
[Here](https://puu.sh/wO232/4e43bafb33.DMP) is my "Microsoft.VSCode.CPP.Extension" dump file.

**Comment by sean-mcmanus** [comment]
The dmp is identical to what an idle state is expected to look like. So no deadlocks, no running threads, and no crashed threads. Can you provide any logs?

**Comment by ArtBlnd** [comment]
Okay, just wait a sec.

**Comment by ArtBlnd** [comment]
my VSCode seems not emits log file... I don't know why. 
maybe something is wrong with my environments.

**Comment by ArtBlnd** [comment]
Tried adding environment variables on `System Environment`, also on cmd with `set` command
just not working.

**Comment by bobbrow** [comment]
What variables did you add?  [These are the variables you need to set](https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/LanguageServer/Enabling%20logging.md).

**Comment by ArtBlnd** [comment]
VSCODE_CPP_LOGDIR=C:\Logs
~~VSCODE_CPP_LOGFILE_LEVEL=9~~ (mod: please don't use 9)
VSCODE_CPP_LOGFILE_LEVEL=7

**Comment by bobbrow** [comment]
Did you create the C:\Logs folder first?  The extension doesn't attempt to create the folder (and wouldn't have permission to create a folder in the root of the C: drive anyway), so it will fail to log if the folder does not already exist.

If the folder is created and you set the variables and didn't get a log file in the C:\Logs folder, then VS Code didn't pick them up.  If you set the variables from the command prompt, you'll need to launch VS Code from the command prompt (unless you used `setx`).  If you set the variables from the Environment Variables UI in Windows, then it should pick them up the next time you launch VS Code from the Start Menu or Taskbar.  Close all open VS Code windows first.

**Comment by bobbrow** [comment]
Oh, and please use log level 7, not 9.  It will make the logs easier for us to read.

**Comment by ArtBlnd** [comment]
I already did all, It doesn't work D:

**Comment by bobbrow** [comment]
Did you use the Windows UI to set the variables?  If Windows is not passing the environment to VS Code, try rebooting to force it globally.

**Comment by Cerh** [comment]
I have a similar situation, but on Ubuntu 16.04. VSСode writes 'Parsing ..'. SQLite files are created.
![11121](https://user-images.githubusercontent.com/11939942/28462884-0dde1b76-6e26-11e7-98bd-841ee861cd2c.png)

**Comment by sean-mcmanus** [comment]
@Cerh The "Parsing..." message is a completely different case from the "Initializing..." message. Can you file a separate issue? It seems like you're getting hung processes or something?

**Comment by nikitablack** [comment]
Hello. I have the same problem. With my main project intellisence doesn't want to work but I can't create a minimal reproducible  example because with small projects all works fine. I don't have any errors in the problems window. I have an icon in bottom right corner saying `Parsing...`. When I hover over an token I get `Loading...` or nothing at all. The log file is huge (more than 200MB) - I'd like to filter it before posting, what kind of information would be helpful for you?

**Comment by bobbrow** [comment]
@nikitablack, what logging level are you using?  Level 5 is sufficient for basic investigation.  Level 7 adds some additional information in case Level 5 doesn't tell us what we want.  Level 9 and above is really only for internal use when debugging specific problems, and is usually too distracting for general triage.  I may push a change to cap the level on release builds to 7 in the next release to avoid creating logs that are too big.

**Comment by nikitablack** [comment]
@bobbrow Here's a log with Level 5.
[vscode.cpp.log.24508.txt](https://github.com/Microsoft/vscode-cpptools/files/1182287/vscode.cpp.log.24508.txt)

**Comment by bobbrow** [comment]
That log is full of SQLite errors.  It would appear that we are unable to create a database on your machine.  Can you try changing the `"browse.databaseFilename"` setting in **c_cpp_properties.json** to another location on your hard disk that you are confident has space and read/write access?

**Comment by sean-mcmanus** [comment]
@nikitablack Another possibility is that multiple processes are trying to access the database. Database operations will fail if you use > 1 instance of VS Code [Insiders] on workspaces that access the same browse database. It's possible dangling processes from old VS Code instances are still using the database.

**Comment by nikitablack** [comment]
@bobbrow It seems I can't find this property - there's no autocompletion for it as well as hint and after setting it no db file created at specified location.

@sean-mcmanus Yes, I restarted machine and everything returned back to normal. Thank you. I should do it before posting.

**Comment by bobbrow** [comment]
@nikitablack it's the property marked `<here>` at the bottom of this example:

```
        {
            "name": "Win32",
            "includePath": [
                "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include",
                "${workspaceRoot}"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE"
            ],
            "intelliSenseMode": "msvc-x64",
            "browse": {
                "path": [
                    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include/*",
                    "${workspaceRoot}"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": "<here>"
            }
        }
```

**Comment by nikitablack** [comment]
There are still issues. The problem I had before appears after some time, so I have to restart my computer over and over again. Moreover, I have errors and I don't know where they come from. For example when I restart vscode and open a project I see the error:
```
file: 'file:/RendererImage.cpp'
message: 'incomplete type is not allowed'
```
for that line:
```
const std::string RendererImage::name("Image Renderer"); //static initialization
```
I don't see squiggles on `#include` directive (there are many includes) and the log file is empty at this point (with level 5). The project itself compiles fine. Also I can't reproduce this in small project.

How else can I track down the problem?

**Comment by sean-mcmanus** [comment]
The issue with it getting stuck on "Initializing..." after a crash should have been fixed in 0.14.1. If there are additional issues, a new issue should be opened.
